source/ota.c

```diff
@@ -3112,6 +3161,16 @@ static void initializeLocalBuffers( void )
     otaAgent.fileContext.pSignature = &sigBuffer;
 }
 
+static void resetEventQueue( void )
+{
+    OtaEventMsg_t eventMsg = { 0 };
+
+    while( otaAgent.pOtaInterface->os.event.recv( NULL, &eventMsg, 0 ) == OtaOsSuccess )
```

Comment:
This could cause a memory leak if the events queued are those of job or data events for which user has created a buffer. We should probably handle the events with a user callback so that application can free those buffers.

Reply:
Call handleUnexpectedEvents to handle that scenarios. And checked that no other events need to be handled.